### DISCUSSION TITLE
How can I pass query parameters and refresh the data?

### DISCUSSION THREAD

**Discussion by zzy-life** [nuxt/nuxt#29347] [Questions]
Reproducible clone: 
https://stackblitz.com/edit/nuxtframeworkdiscussions2568-gg1k1k?file=app.vue

F5 refresh is required in the code to refresh the data
Correct query url:
https://api.webarcx.com/seoapi/api/get/product/1/10/2116?creator=%E8%BD%AF%E8%91%97%E6%A8%A1%E6%9D%BF

**Comment by zzy-life** [comment]
You must set external to true to load normally,
<NuxtLink :external="true"
</NuxtLink>

**Comment by xjccc** [comment]
```
<script setup>
const route = useRoute();
const creator = computed(() => route.query.creator);
const { data, status, refresh } = useFetch(
  'https://api.webarcx.com/seoapi/api/get/product/1/10/2116',
  {
    query: {
      creator
    },
  }
);

console.log(status);
// this changes the query params, but the data remains stale
const router = useRouter();
</script>

<template>
  <div>
    <div>URL: {{ decodeURIComponent(route.fullPath) }}</div>
    <div>query: {{ decodeURIComponent(creator) }}</div>
    <NuxtLink :to="{ name: '', query: {} }">Trigger</NuxtLink>
    <NuxtLink :to="{ name: '', query: { creator: '软著模板' } }"
      >Trigger</NuxtLink
    >
    <NuxtLink :to="{ name: '', query: { creator: '软著意义' } }"
      >Trigger</NuxtLink
    >
    <pre>
    {{ data }}
    </pre>
  </div>
</template>

```

**Reply by zzy-life** [reply]
> ```
> <script setup>
> const route = useRoute();
> const creator = computed(() => route.query.creator);
> const { data, status, refresh } = useFetch(
>   'https://api.webarcx.com/seoapi/api/get/product/1/10/2116',
>   {
>     query: {
>       creator
>     },
>   }
> );
> 
> console.log(status);
> // this changes the query params, but the data remains stale
> const router = useRouter();
> </script>
> 
> <template>
>   <div>
>     <div>URL: {{ decodeURIComponent(route.fullPath) }}</div>
>     <div>query: {{ decodeURIComponent(creator) }}</div>
>     <NuxtLink :to="{ name: '', query: {} }">Trigger</NuxtLink>
>     <NuxtLink :to="{ name: '', query: { creator: '软著模板' } }"
>       >Trigger</NuxtLink
>     >
>     <NuxtLink :to="{ name: '', query: { creator: '软著意义' } }"
>       >Trigger</NuxtLink
>     >
>     <pre>
>     {{ data }}
>     </pre>
>   </div>
> </template>
> ```

Thank you so much!

**Reply by zzy-life** [reply]
> ```
> <script setup>
> const route = useRoute();
> const creator = computed(() => route.query.creator);
> const { data, status, refresh } = useFetch(
>   'https://api.webarcx.com/seoapi/api/get/product/1/10/2116',
>   {
>     query: {
>       creator
>     },
>   }
> );
> 
> console.log(status);
> // this changes the query params, but the data remains stale
> const router = useRouter();
> </script>
> 
> <template>
>   <div>
>     <div>URL: {{ decodeURIComponent(route.fullPath) }}</div>
>     <div>query: {{ decodeURIComponent(creator) }}</div>
>     <NuxtLink :to="{ name: '', query: {} }">Trigger</NuxtLink>
>     <NuxtLink :to="{ name: '', query: { creator: '软著模板' } }"
>       >Trigger</NuxtLink
>     >
>     <NuxtLink :to="{ name: '', query: { creator: '软著意义' } }"
>       >Trigger</NuxtLink
>     >
>     <pre>
>     {{ data }}
>     </pre>
>   </div>
> </template>
> ```
There is another problem. Now the data is updated, but the page view does not change. The expected result is that NuxtLink can navigate me to a new page and get the data to refresh the view.
I have updated the clone
